### ISSUE TITLE
Is it possible to extend this from fixed width / height

### ISSUE THREAD

**Issue by jimiero** [cmextension/cmavatar#6]
Hey guys, nice plugin, I was wondering if you think you can also add option where you can select different width / height images, and keep proportion, because currently seams all avatars have square layout.

**Comment by tranduyhung** [comment]
The avatar is not always square, it is in the same ratio to the original image's one. You can try to upload a rectangle image to see that.

**Comment by jimiero** [comment]
Look what problem I have when I update a image
![screen shot 2015-10-27 at 10 28 51 am](https://cloud.githubusercontent.com/assets/3626762/10752847/8ce09aea-7c95-11e5-8d57-fbc3b4416478.png)

**Comment by jimiero** [comment]
@tranduyhung looking at the source code I see this:

```
<div class="control-group">
        <img src="/my-joomlasubfolder/&lt;div class=" cmavatar"="">
<img src="/my-joomlasubfolder/images/avatars/0f2c7761d25a5c962857be4eb8861270.jpg">
</div>
```

**Comment by tranduyhung** [comment]
@jimiero What is your CMAvatar version? I don't see this problem in the latest 1.1.1 version.

![2015-10-27-154201_682x343_scrot](https://cloud.githubusercontent.com/assets/503262/10753104/5a29e4ae-7cc1-11e5-9587-8b92108b9859.png)

**Comment by jimiero** [comment]
@tranduyhung Is cmavatar-1.1.1 downloaded from github

**Comment by tranduyhung** [comment]
I just downloaded from Github and tried but couldn't reproduce this problem. I will take a look at this and release a new version to fix it.

**Comment by jimiero** [comment]
@tranduyhung problem sorted, the issue was caused by another plugin, remove that plugin and it's fine now

**Comment by jimiero** [comment]
@tranduyhung sorry for bugging you but still the problem persists even after that plugin was removed.

Another thing, I see that pictures which has transparent background they lose the transparent background when you upload and gets black, do you think you can sort that on the next version ?

**Comment by tranduyhung** [comment]
At the present the plugin uses JPG as the result avatar. Keep PNG's transperency and keep GIF's animation are possible to add to the plugin but require lots of work and some other third party tools or PHP extensions (like ImageMagick for example). I will add them when I have time.

**Comment by jimiero** [comment]
@tranduyhung do you have a skype id ?

Also here are some improvements suggestions:
- show a default avatar if user has no avatar set
- Keep PNG's transperency and keep GIF's 

Let me know if we can talk on skype

**Comment by tranduyhung** [comment]
Sorry @jimiero I don't use Skype, you can post your suggestions here on Github. Thanks!

**Comment by tranduyhung** [comment]
@jimiero your problem was fixed in 1.2.0 branch, transparent PNG and GIF are also supported now. You can download and test 1.2.0 branch, please create issues if you find any problems.
Animated GIF is still not supported, I will consider to add it in future.
